docs/explanations/default-options.rst

```diff
@@ -0,0 +1,63 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _explanations-default-options:
+
+*********************
+Default build options
+*********************
+
+Meson offers many `built-in options <https://mesonbuild.com/Builtin-options.html>`__,
+and in the vast majority of cases those have good defaults. There are a couple
+of cases however where ``meson-python`` either is forced or chooses to override
```

Comment:
Forced?

Reply:
I'll rephrase as "needs to" if that's better. Overrides for, e.g., `--prefix` are necessary.

Reply:
Technically, they are not necessary. We could very well use the default prefix and use the `$DESTDIR` environment variable to install in a temporary directory. I would even argue that this would be better. Anyhow, it does not matter.

Reply:
Yes, I think it was only needed for the Debian heuristics, I think we can drop it now that Meson uses directory placeholders for all paths.